packages/slice-machine/lib/mock/CustomType.ts

```diff
@@ -1,88 +1,54 @@
+  const widgetsConfig = widgets.reduce<
+    Partial<Record<WidgetKey, DocWidgetMockConfig>>
+  >((acc, [key, w]) => {
+    const legacyFieldConfig: Partial<Record<string, unknown>> | undefined =
+      legacyMockConfig[key];
+    if (!legacyFieldConfig) return acc;
+
+    const widgetConfig = buildWidgetMockConfig(w, legacyFieldConfig);
+    if (!widgetConfig) return acc;
+
+    return { ...acc, [key]: widgetConfig };
+  }, {});
+
+  return { value: widgetsConfig };
 }
 
-const fieldsHandler = handleFields(Widgets);
-
-const groupHandler = (group: GroupSM, mockConfig: CustomTypeMockConfig) => {
-  const items = [];
-  const entries = group.config?.fields;
-  for (let i = 0; i < Math.floor(Math.random() * 6) + 2; i++) {
-    items.push(fieldsHandler(entries, mockConfig));
-  }
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-  return items;
-};
-
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const sliceZoneHandler = () => {};
-
-const createEmptyMock = (type: string) => ({
-  id: faker.datatype.uuid(),
-  uid: null,
-  type,
-  data: {},
-});
-
-// eslint-disable-next-line @typescript-eslint/require-await
-export default async function MockCustomType(
+export default function MockCustomType(
   model: CustomTypeSM,
-  mockConfig: CustomTypeMockConfig
+  legacyMockConfig: CustomTypeMockConfig
 ) {
-  const customTypeMock: Mock = createEmptyMock(model.id);
-  const maybeUid = model.tabs.reduce((acc, curr) => {
-    const maybeWidgetUid = Object.entries(curr.value).find(
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-unsafe-member-access
-      ([_, e]) => e.value.type === "UID"
-    );
-    if (!acc && maybeWidgetUid) {
-      return curr;
-    }
-    return acc;
-  });
-
-  if (maybeUid) {
-    customTypeMock.uid = Widgets.UID.handleMockConfig();
-  }
-
-  for (const [, tab] of Object.entries(model.tabs)) {
-    const { fields, groups, sliceZone } = Tab.organiseFields(tab);
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const mockedFields = fieldsHandler(fields as [], mockConfig);
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    customTypeMock.data = {
-      ...customTypeMock.data,
-      ...mockedFields,
-    };
-    groups.forEach((group) => {
-      const { key, value } = group;
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-assignment
-      const groupMockConfig = CustomTypeMockConfig.getFieldMockConfig(
-        mockConfig,
-        key
-      );
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      const groupFields = groupHandler(value, groupMockConfig);
-      customTypeMock.data[key] = groupFields;
-    });
-
-    if (sliceZone) {
-      sliceZoneHandler();
-    }
-  }
-  return customTypeMock;
+  const prismicModel = CustomTypes.fromSM(model);
+  const documentMockConfig = buildDocumentMockConfig(
```

Comment:
I'm not very familiar with mocks tbh, does this helper also generate the mocks for the slice that are inside the custom type ?

What I mean is, if there are slices attached to the custom type, will their own custom config be ignore or would it be included ?

Reply:
I kept the same behavior, right now you can generate either mocks for shared slices or for custom type but the slices inside the custom type don't have mocks, it's an empty array that is returned.
But, The lib can generate mocks for the slice inside the custom type as long as you also provide shared slice models along with the one from the custom type.
But there is the question wether or not we want to generate or just concatenate mocks already generated for each slice.
Anyway, the solution was to keep what we have today, meaning an empty array. And ultimately we don't use custom types mocks.

Reply:
Yeah i was thinking about concat them too also.
But i guess it should be done here but just when we need it for whatever apps we need it.

Reply:
It was not supported before and it's an enhancement that we might consider at some point we I don't think that should be the focus right now considering that this feature is basically useless since we don't have a similar simulator for custom types.
Also it'll be a new behavior to introduce on SM so it'll be probably be up to the SM team to evolve those in the future depending on the needs.